docs/server.rst

```diff
@@ -0,0 +1,54 @@
+.. _server:
+
+=======================
+Use with Jupyter Server
+=======================
+
+This page describes the way `Jupyter Server <https://github.com/jupyter/jupyter_server>`_ uses the Kernel Management module.
```

Comment:
Ultimately this page probably belongs in the Jupyter Server docs, but if it makes sense to put it here at the moment, I don't have a problem with that.

Reply:
I will work next week with @Zsailer on the jupyter server doc. If this is merged before, we can reshuffle the details and also do a second pass on the kernel mgmt docs.